src/arrays.js

```diff
@@ -8,33 +8,104 @@ const each = (elements, cb) => {
+    // the cb invocation is given in underscore.  No directions are given regarding following underscore, only that this exercise, or rather, just "something"
+    // is based off of underscorejs
+  }
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  const out = [];
+  each(elements, (el) => {
+    out.push(cb(el));
+  });
+  return out;
 };
 
 const reduce = (elements, cb, memo = elements.shift()) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  each(elements, (el) => {
+    memo = cb(memo, el);
+  });
+  return memo;
 };
 
 const find = (elements, cb) => {
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
+  //
+  //
+  // REVIEW PREVIOUS COMMIT VERSION USING BUILT EACH FUNCTION AND BRING UP DURING A Q & A
+  //
+  //
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i])) {
+      return elements[i];
+    }
+  }
+  return undefined;
 };
 
 const filter = (elements, cb) => {
   // Similar to `find` but you will return an array of all elements that passed the truth test
   // Return an empty array if no elements pass the truth test
+  const out = [];
+  each(elements, (el) => {
+    if (cb(el) === true) out.push(el);
+  });
+  return out;
 };
 
 const flatten = (elements) => {
   // Flattens a nested array (the nesting can be to any depth).
   // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
+
+// we want to recursively flatten layers
+/*
+ * establish base case
+ * if array has no arrays as elements return array (test for 'has or has not' involves iteration)
+ * create master array
+ * for each element in array
+ *   combine the result of flatten array to master array
+ * return flatten on master array
+ *
+ * alternative
+ *
+ *
+ *
+ * the recursive call will return a number while base case returns an array
+ */
+  // easiest case [1] =>  [1]
+  // next easiest case [1, [2,2,3]] etc. => [1,2]
+  // [1]
+  // [1,[2]]
+  // create empty output array // []
+  // if the current input with var name array is not an array
+  //   assign output array to [input]
+  //   return output array
+  // iterate over inputted array(for each) // elements = [2,2,3], curr = 2 out2 = [2,2,3]
+  //   concatenate to output array the return value of running flatten for the current input // output = [1]
+  // return output array
+  //
+  // new stack
+  // iterate over [2,2,3]
+  //
+  // [1,[2]]
+  let newArr = []; // this is what the function will at the end return
```

Comment:
Your implementation works. Just thought I'd point you to a much terser implementation so you can get some ideas of other ways this function could be implemented. 

Here's an example implementation that makes use of the `reduce` function that was implemented earlier:
```
const flatten = (array) => {
    return reduce(array, (a, b) => a.concat(Array.isArray(b) ? flatten(b) : b), []);
};
```
Probably not a very intuitive implementation, so let's break this down. Recall that `reduce` takes a list of elements, a callback to pass those elements to, and a memo, which is just an empty array in this case. It's the same as how you initialized a new array at the beginning of your function. So we're passing our initial array to reduce as its first argument, and then our callback checks that the next element we're looking at is an array or not. If so, we continue to recursively call this `flatten` function, so that we can strip away every level of nesting, until we finally get an element that is not an array, and then concatenate that onto our memo array.